### ISSUE TITLE
Old implementations

### ISSUE THREAD

**Issue by bbrala** [json-api/json-api#1313]
I've clicked around in the implementations in the documentation and it seems there is quite a lot of project that haven't been updated in ages. For the PHP side i see quite a few that one could argue are abandoned. 

Would you mind if i went through and a least do some sorting so active projects are on top? This way when users click through the list of implementations they see relatively active projects first. I'm not sure if assuming a project dead is really feasable since that sometimes is rather hard to know for sure.

**Comment by ethanresnick** [comment]
I'd like to do some sorting too, as the list is much less useful than it could be because it doesn't highlight the most-developed implementations. The trick is finding some general criteria to apply to sort things in a way that people feel is fair. 

I can think of one thing off the top of my head that should be uncontroversial: if an implementation is still based on [RC3](http://jsonapi-rc3.herokuapp.com/) of the spec, it should definitely be removed. That criterion is easy to check -- does the implementation have separate `attributes` and `relationships` keys in its resource objects? --  and would mean that the implementation hasn't been updated in years.

Maybe you can post a list here of all the implementations that are still based on RC3 and we'll start by killing those? Then, if there's still too much cruft in the implementations list, we can explore other means of sorting what's left?

cc @dgeb 
ref #733 #1151

**Comment by wimleers** [comment]
Oh interesting, first time I hear about RC3 and `attributes` <> `relationships` not yet being a thing there!

Sounds like a great criterion for at least a first round of clean-up!

**Comment by ethanresnick** [comment]
Yup. @wimleers, if you have some time to help, maybe one of you could take the PHP server implementations and the other the client implementations and we'll start there?

**Comment by wimleers** [comment]
I'll keep this issue in mind/this browser tab open for when I do have some down time :)

**Comment by bbrala** [comment]
Yeah, same, please mention here should you start and if you are starting on the client or server list.

Personally i think we should (at least for php) also look at how well supported it is, i mean, when i see Laravel 5.2 support as the max, that is like a few years old, and no longer maintained, which would mean the package is kind of worthless right now.

**Comment by bbrala** [comment]
The PHP clients seem fine actually, i just made a small PR to order them by name.

**Comment by wimleers** [comment]
I've been looking at many implementations today, and _many_ of them are pretty simplistic and incomplete. Implementing the absolute basics of the JSON:API spec is easy, but also including support for collection filtering, sorting, pagination for example is usually lacking.

I'm not sure how we can surface this in easy-to-scan format though. Thoughts?

**Comment by bbrala** [comment]
Perhaps something like a feature table would work. But I fear that will be outdated pretty fast.

**Comment by bbrala** [comment]
A lot of the features you mentioned aren't compulsory though. So perhaps that might be a separation. Hmm.

**Comment by ethanresnick** [comment]
> I'm not sure how we can surface this in easy-to-scan format though. Thoughts?

> Perhaps something like a feature table would work. But I fear that will be outdated pretty fast.

Maybe a feature table, but it's is the job of each implementation to maintain their feature table by submitting PRs? I feel like that would make sense. Then we could use a bit of JS on the frontend to enable people to sort/filter by supported features.

**Comment by wimleers** [comment]
> I feel like that would make sense. Then we could use a bit of JS on the frontend to enable people to sort/filter by supported features.

👍
